lib/segment.ex

```diff
@@ -10,6 +10,40 @@ defmodule Segment do
     Agent.start_link(fn -> %{endpoint: endpoint, key: key} end, name: __MODULE__)
   end
 
+  @doc """
+  The child specifications
+
+  ## Examples
+
+    iex> Segment.child_spec([key: "something"])
+    %{
+      id: Segment,
+      start: {Segment, :start_link, ["something", nil]}
+    }
+
+    iex> Segment.child_spec([])
+    ** (KeyError) key :key not found in: []
+
+    iex> Segment.child_spec([key: "something", endpoint: "http://example.com"])
+    %{
+      id: Segment,
+      start: {Segment, :start_link, ["something", "http://example.com"]}
+    }
+
+    iex> Segment.child_spec([key: "something", endpoint: "http://example.com"])
+    %{
+      id: Segment,
+      start: {Segment, :start_link, ["something", "http://example.com"]}
+    }
+
+  """
+  def child_spec(arg) do
+    %{
+      id: Segment,
+      start: {Segment, :start_link, [Keyword.fetch!(arg, :key), Keyword.get(arg, :endpoint)]}
```

Comment:
From what I understood `Keyword.fetch!/2` is used for required parameters and `Keyword.get/2` for optional parameters.

I'd suggest to move to outside the map the make more clear which one is required and optional. And also the make easier to find the error in the error stack when the `key` is not supplied. 

For example:
```elixir
key = Keyword.fetch!(arg, :key)
endpoint = Keyword.get(arg, :endpoint)

%{
  id: Segment,
  start: {Segment, :start_link, [key, endpoint]}
}
```

or 

```elixir
params = [
  Keyword.fetch!(arg, :key),
  Keyword.get(arg, :endpoint)
]

%{
  id: Segment,
  start: {Segment, :start_link, params}
}
```